gfx.cpp

```diff
@@ -9402,7 +9402,15 @@ class GfxInternal
         return kGfxResult_NoError;
     }
 
-    bool transitionResource(Buffer &buffer, D3D12_RESOURCE_STATES resource_state, bool implicit_allowed = false)
+    enum TransitionType
+    {
+        kTransitionType_Implicit = 0,
+        kTransitionType_Explicit,
+
+        kTransitionType_Count
+    };
+
+    bool transitionResource(Buffer &buffer, D3D12_RESOURCE_STATES resource_state, TransitionType implicit_allowed = kTransitionType_Explicit)
```

Comment:
We could possibly rename from `implicit_allowed` to `transition_type`.
As-is, we're essentially having `if(implicit_allowed) -> implicit is not allowed` 🙂